src/data/list/lexicographic.lean

```diff
@@ -0,0 +1,159 @@
+/-
```

Comment:
We can just call this file `lex.lean`, right? I think it's better when the filenames match the declaration names, and it's shorter :)

Otherwise LGTM. Thanks!
bors d+